app/web/api/autonomous_agents/views.py

```diff
@@ -0,0 +1,103 @@
+import asyncio
+import ctypes
+import json
+import sys
+from asyncio import sleep
+from concurrent.futures import ThreadPoolExecutor
+
+from scripts.file_operations import write_to_file, read_file
+
+sys.path.append("scripts/")
+# from app.web.api.autonomous_agents.schema import AutonomousAgent
+from scripts.main import construct_prompt, AutonomousAgent
+
+import cmd
+
+from tkinter.ttk import Style
+
+from starlette.responses import JSONResponse
+from typing import Dict
+from fastapi import APIRouter
+from scripts.ai_config import AIConfig
+from scripts.config import Config
+from scripts.memory import PineconeMemory, get_memory
+import threading
+
+router = APIRouter()
+
+thread_dict = {}  # Dictionary to store the threads
+
+
+@router.post("/autonomous_agents")
+async def create_autonomous_agent(body: Dict):
+    ai_config = AIConfig(body["ai_name"], body["ai_role"], body["ai_goals"], interactive_input=False)
+    ai_config.save()
+
+    return JSONResponse(
+        content={"message": "Autonomous agent created", "id": 1})  # only one main agent is supported for now
+
+@router.patch("/autonomous_agents/{id}")
+async def start_autonomous_agent(body: Dict):
+    if body["status"] == 'active':  # this is the only status that is supported for now
+        cfg = Config()
+        cfg.memory_backend = "no_memory"
+        cfg.set_continuous_mode(True)  # this endpoint only supports continuous mode for now
+        memory = get_memory(cfg, init=True)
+
+        prompt = construct_prompt()
+        autonomous_agent = AutonomousAgent(
+            cfg=cfg,
+            prompt=prompt,
+            memory=memory,
+            user_input="Determine which next command to use, and respond using the format specified above:"
+        )
+
+        # Start the loop in a separate thread
+        thread = threading.Thread(target=autonomous_agent.start_loop)
+        try:
+            thread.start()
+        except SystemExit:
+            # when the agent shuts down after task completion, it returns a SystemExit exception
+            pass
+
+        # Get the thread ID
+        thread_id = thread.ident
+        thread_dict[thread_id] = thread  # Store the thread in the dictionary
+        return JSONResponse(content={"message": "Autonomous agent created", "thread_id": thread_id})
+    elif body["status"] == 'inactive':
+        if "thread_id" not in body:
+            return JSONResponse(content={"message": "thread_id is required in the request body"}, status_code=400)
+        thread_id = body["thread_id"]
+        # Terminate the thread using the thread ID
+        thread = thread_dict.get(thread_id)  # Get the thread from the dictionary
+        if thread:
+            if not thread.is_alive():
+                return JSONResponse(content={"message": "The thread is already terminated"}, status_code=500)
+
+            threads_to_terminate = read_file("threads_to_terminate.json")
```

Comment:
reading and json.loading the threads_to_terminate.json could be in a function, it's also used in the `def termination_check`